### PR TITLE
Liquibase support and repo refactor

### PR DESCRIPTION
**Pull request by vfilimo**

### CODE REVIEW

File: src/main/java/online/store/book/service/BookServiceImpl.java [hunk 1]
```diff
@@ -23,18 +23,32 @@ public BookDto save(CreateBookRequestDto bookRequestDto) {
...
-        Book book = bookRepository.findBookById(id).orElseThrow(
+        Book book = bookRepository.findById(id).orElseThrow(
                 () -> new EntityNotFoundException("Can't find book with id: " + id)
         );
         return bookMapper.toDto(book);
     }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto bookRequestDto) {
+        findBookById(id);
```
Comment (sarakhmen) [inline src/main/java/online/store/book/service/BookServiceImpl.java]: why do you have this method call here? Seems to be redundant. Fix in all places

File: src/main/java/online/store/book/service/BookServiceImpl.java [hunk 2]
```diff
@@ -23,18 +23,32 @@ public BookDto save(CreateBookRequestDto bookRequestDto) {
...
+        Book bookEntity = bookMapper.toEntity(bookRequestDto);
```
Comment (sarakhmen) [inline src/main/java/online/store/book/service/BookServiceImpl.java]: We recommend making all request mappings at the controller level. Fix in all places

**Review by sarakhmen — CHANGES_REQUESTED** [review]
let's improve your solution :)

File: src/main/java/online/store/book/service/BookServiceImpl.java [hunk 3]
```diff
@@ -23,18 +23,34 @@ public BookDto save(CreateBookRequestDto bookRequestDto) {
...
-        Book book = bookRepository.findBookById(id).orElseThrow(
+        Book book = bookRepository.findById(id).orElseThrow(
                 () -> new EntityNotFoundException("Can't find book with id: " + id)
         );
         return bookMapper.toDto(book);
     }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto bookRequestDto) {
+        bookRepository.findById(id).orElseThrow(
```
Comment (Elena-Bruyako) [inline src/main/java/online/store/book/service/BookServiceImpl.java]:
```suggestion
        Book book = bookRepository.findById(id).orElseThrow(
```

File: src/main/java/online/store/book/service/BookServiceImpl.java [hunk 4]
```diff
@@ -23,18 +23,34 @@ public BookDto save(CreateBookRequestDto bookRequestDto) {
...
+                () -> new EntityNotFoundException("Can't find book with id: " + id));
+        Book bookEntity = bookMapper.toEntity(bookRequestDto);
+        bookEntity.setId(id);
+        return bookMapper.toDto(bookRepository.save(bookEntity));
```
Comment (Elena-Bruyako) [inline src/main/java/online/store/book/service/BookServiceImpl.java]:
```suggestion
        bookMapper.updateBookFromDto(bookRequestDto, book);
        return bookMapper.toDto(bookRepository.save(book));
```

File: src/main/java/online/store/book/service/BookServiceImpl.java [hunk 5]
```diff
@@ -23,18 +23,34 @@ public BookDto save(CreateBookRequestDto bookRequestDto) {
...
+    }
+
+    @Override
+    public void delete(Long id) {
+        bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find book with id: " + id));
```
Comment (Elena-Bruyako) [inline src/main/java/online/store/book/service/BookServiceImpl.java]:
```suggestion
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**